tensorpipe/channel/context.h

```diff
@@ -25,6 +25,7 @@ namespace channel {
 // context. All registered instances are assumed to be eligible
 // channels for all pairs.
 //
+template <typename TTensor>
 class Context {
```

Comment:
As I said elsewhere, I think it would be nice if we could add this at the bottom of this file:
```
using CpuContext = Context<CpuTensor>;
using CudaContext = Context<CudaTensor>;
```
as shorthands since we'll be using them often.